exercises/simple-cipher/.meta/src/reference/java/Cipher.java

```diff
@@ -32,23 +32,29 @@ public String getKey() {
     }
 
     public String encode(String plainText) {
-        String copy = plainText;
         StringBuilder ciphertext = new StringBuilder(plainText.length());
+        String sameLengthKey = getKeySameLengthAsPlaintext(plainText, key);
 
-        while (copy.length() > 0) {
-            for (int i = 0; i < Math.min(copy.length(), key.length()); i++) {
-                ciphertext.append(encodeCharacter(copy, i));
-            }
-            copy = "";
-            for (int i = ciphertext.length(); i < plainText.length(); i++) {
-                copy = copy + plainText.charAt(i);
-            }
+        for (int index = 0; index < plainText.length(); index++) {
+            ciphertext.append(encodeCharacter(plainText, index, sameLengthKey));
         }
 
         return ciphertext.toString();
     }
 
-    private char encodeCharacter(String plainText, int index) {
+    private String getKeySameLengthAsPlaintext(String plainText, String key) {
+
+        String copy = key;
```

Comment:
It might be better to use a [StringBuilder](https://docs.oracle.com/javase/9/docs/api/java/lang/StringBuilder.html) here as we're doing possibly quite a few appends? And in that case naming it something like `longKeyBuilder` might be more descriptive? 🙂